problems/5-memes/problem.md

```diff
@@ -1,22 +1,53 @@
 # Memes for my eyes
 
-Any meme you upload is almost instantly available for retrieval. Other files also. Content is made available by root CID and by CAR shard CID. In this exercise you are going to fetch a text file from the public gateway by it's root CID and log it to the console.
+Ain't you cool? Now, let's shift gears and dive into the next challenge: retrieving content from a decentralized network.
 
-You can use the `w3s.link` gateway to retrieve the data but the magic of decentralized peer to peer networking means that you should be able to use any gateway.
+## The Challenge:
+Every meme (or any other file) you upload is instantly available for retrieval. In this exercise, your task is to fetch a text file from the public gateway using its root CID. The cool part? You can use any gateway, thanks to the decentralized nature of peer-to-peer networking. But for simplicity, we’ll use the w3s.link gateway.
 
-Create a _new_ file for your solution e.g. `ex6.mjs`.
+## What You Need to Do:
+1. Start by creating a new JavaScript file, ex6.mjs. 
 
-To retrieve data from a gateway you should format the request like:
+2. Fetch the CID from `stdin`: The CID (Content Identifier) of the file you need to fetch will be provided via `process.stdin`. You’ll need to read this input first.
+```js
+import fs from 'node:fs'
+import { CID } from 'multiformats'
 
+const cid = CID.decode(fs.readFileSync(process.stdin.fd))
 ```
+3. Construct the Gateway URL: Format the URL to fetch the content using the w3s.link gateway. The URL format is:
+```js
 https://w3s.link/ipfs/<CID>
 ```
+Replace <CID> with the actual CID you decoded.
+```js
+const url = `https://w3s.link/ipfs/${cid}`
+```
+4. Fetch the Content: Use the fetch API to send an HTTP request to the gateway and retrieve the data.
+```js
+const res = await fetch(url)
+```
+5. Output the Content: Finally, write the fetched content directly to `process.stdout`.
+```js
+process.stdout.write(await res.text())
+```
+
+## What’s Happening Here?
+**CID Decoding:** The CID you receive is in a compact binary form. You use the multiformats library to decode it into a human-readable string that you can use in the URL.
+**Gateway Flexibility:** While we use w3s.link here, you can use any gateway that supports IPFS (InterPlanetary File System).
```

Comment:
They are typically called IPFS gateways - they'r enot a gateway for anything else.

```suggestion
**Gateway Flexibility:** We use w3s.link here, but you can use any IPFS gateway.
```